src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -0,0 +1,159 @@
+} from '../../../common/error/errors';
+import { RemoteCheckoutSynchronizationError } from '../../../remote-checkout/errors';
+
+import {
+    ButtonColor,
+    ButtonType,
+    EnvironmentType,
+    GooglePaymentsError,
+    GooglePaymentData,
+    GooglePayAddress,
+    GooglePayClient,
+    GooglePayInitializer,
+    GooglePayPaymentDataRequestV1,
+    GooglePayScriptLoader,
+    GooglePaySDK,
+    TokenizePayload
+} from './';
+
+export default class GooglePayPaymentProcessor {
+    private _googlePaymentsClient!: GooglePayClient;
+    private _methodId!: string;
+    private _googlePaymentDataRequest!: GooglePayPaymentDataRequestV1;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _googlePayScriptLoader: GooglePayScriptLoader,
+        private _googlePayInitializer: GooglePayInitializer,
+        private _billingAddressActionCreator: BillingAddressActionCreator
+    ) { }
+
+    initialize(methodId: string): Promise<void> {
+        this._methodId = methodId;
+
+        return this._configureWallet();
+    }
+
+    deinitialize(): Promise<void> {
+        return this._googlePayInitializer.teardown();
+    }
+
+    createButton(callback: () => {}): HTMLElement {
+        return this._googlePaymentsClient.createButton({
+            buttonColor: ButtonColor.Default,
+            buttonType: ButtonType.Short,
+            onClick: callback,
+        });
+    }
+
+    updateBillingAddress(billingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors> {
+        if (!this._methodId) {
+            throw new RemoteCheckoutSynchronizationError();
+        }
+
+        const remoteBillingAddress = this._store.getState().billingAddress.getBillingAddress();
+
+        if (!remoteBillingAddress) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const googlePayAddressMapped: BillingAddressUpdateRequestBody = this._mapGooglePayAddressToBillingAddress(billingAddress, remoteBillingAddress.id);
+
+        return this._store.dispatch(
+            this._billingAddressActionCreator.updateAddress(googlePayAddressMapped)
+        );
+    }
+
+    displayWallet(): Promise<GooglePaymentData> {
+        if (!this._googlePaymentsClient && !this._googlePaymentDataRequest) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._googlePaymentsClient.isReadyToPay({
+            allowedPaymentMethods: this._googlePaymentDataRequest.allowedPaymentMethods,
+        }).then( response => {
+            if (response.result) {
+                return this._googlePaymentsClient.loadPaymentData(this._googlePaymentDataRequest)
+                    .then(paymentData => paymentData)
+                    .catch((err: GooglePaymentsError) => {
+                        throw new Error(err.statusCode);
+                    });
+            } else {
```

Comment:
Is needed because the method `isReadyTo Pay` is to validate if GooglePay is ready, and it returns a boolean value, if `response.result` is false, we need to throw an exception to notify that, otherwise we try to load the widget and maybe in this part the widget can crash

Reply:
@clopezh I think the argument is if all the `else` is doing is `throwing` the exception then why not just let that happen on it's own. e.g., if you look at the below happy path will return the value and un-happy path will throw the exception, it's just not bundled in an `else`


```
            if (response.result) {
                return this._googlePaymentsClient.loadPaymentData(this._googlePaymentDataRequest);
            }

            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
```